### PR TITLE
test: 23.7% coverage of add-edit-per-diem page

### PR DESCRIPTION
**Pull request by suyashpatil78**
### <samp>🤖 Generated by Copilot at bc8677d</samp>

This pull request adds new test cases and mock data for the add-edit-per-diem page, which allows users to create or edit per diem expenses. It creates a separate spec file for the test cases, a mock data file for per diem custom inputs, and another mock data file for unflattened per diem expenses. It also modifies the existing setup spec file to import and run the new test cases.

### <samp>🤖 Generated by Copilot at bc8677d</samp>

> _Mock data for forms_
> _`add-edit-per-diem` tests_
> _Autumn leaves falling_

### <samp>🤖 Generated by Copilot at bc8677d</samp>

*  Add mock data for per diem custom inputs and unflattened expense ([link](https://github.com/fylein/fyle-mobile-app/pull/2263/files?diff=unified&w=0#diff-9f195d091788e9bb0c8df097ef4fcffbee53b84e408eda5858d11b718eb2d7c4R1-R92), [link](https://github.com/fylein/fyle-mobile-app/pull/2263/files?diff=unified&w=0#diff-db737e50777751787c68eda6c5be391ed672eb00607a92de6aaf9c945a3010d6R561-R581))
*  Create new spec file for add-edit-per-diem page logic and functionality ([link](https://github.com/fylein/fyle-mobile-app/pull/2263/files?diff=unified&w=0#diff-e07d5851dc6ce6b9d4929a5a7cd5cce88c19fb320d9efbd769f8f1911bbd440dR1-R281))
*  Import and call TestCases2 function from new spec file in setup spec file ([link](https://github.com/fylein/fyle-mobile-app/pull/2263/files?diff=unified&w=0#diff-113c190c8d5e8c3f8c98cf1e699be2562159cd2ba8d04e2e19a31ef5b74d35c2R39), [link](https://github.com/fylein/fyle-mobile-app/pull/2263/files?diff=unified&w=0#diff-113c190c8d5e8c3f8c98cf1e699be2562159cd2ba8d04e2e19a31ef5b74d35c2R283))

## Clickup
https://app.clickup.com/t/85ztk93dz

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/fyle/add-edit-per-diem/add-edit-per-diem-2.page.spec.ts [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+        mockEtxn.tx.org_category_id = undefined;
+        component.etxn$ = of(mockEtxn);
+        categoriesService.getAll.and.returnValue(of([orgCategoryData]));
+        component.getCustomInputs().subscribe((res) => {
+          expect(customInputsService.getAll).toHaveBeenCalledOnceWith(true);
+          expect(component.getPerDiemCategories).toHaveBeenCalledTimes(1);
+          expect(customFieldsService.standardizeCustomFields).toHaveBeenCalledOnceWith([], expenseFieldResponse);
+          expect(customInputsService.filterByCategory).toHaveBeenCalledOnceWith(expenseFieldResponse, 38912);
+          const expenseFieldWithoutControl = res.map(({ control, ...otherProps }) => ({ ...otherProps }));
+          const expectedExpenseFieldWithControl = perDiemCustomInputsData1.map(({ control, ...otherProps }) => ({
```
Comment (OmkarJ13) [inline src/app/fyle/add-edit-per-diem/add-edit-per-diem-2.page.spec.ts:265]: do you mean `expectedExpenseFieldWithoutControl`?

Comment (OmkarJ13) [reply src/app/fyle/add-edit-per-diem/add-edit-per-diem-2.page.spec.ts:265]: Also, should we move the data being transformed via map to mock data?

Comment (suyashpatil78) [reply src/app/fyle/add-edit-per-diem/add-edit-per-diem-2.page.spec.ts:265]: Yes will do that!

File: src/app/core/mock-data/unflattened-expense.data.ts [hunk 2]
```diff
@@ -558,3 +558,24 @@ export const draftUnflattendedTxn3 = {
     fyle_category: 'UNSPECIFIED',
   },
 };
+
+export const unflattenedTxnDataPerDiem = {
+  tx: {
+    skip_reimbursement: false,
+    source: 'MOBILE',
```
Comment (Chethan-Fyle) [inline src/app/core/mock-data/unflattened-expense.data.ts:565]: Enum for source and state as well.

**Review by Chethan-Fyle — APPROVED** [review]

**Pull request merged**